sdk/program/src/lib.rs

```diff
@@ -1,3 +1,650 @@
+//! some native programs cannot be invoked by other programs, a Solana program
+//! may need to verify that certain accounts represent those native programs,
+//! e.g. when verifying that the ed25519 precompile verified a signature.
+//!
+//! For many built-ins, this crate also defines enums that represent the
+//! instructions they process, and constructors for building the instructions.
+//! Some built-ins' instructions and constructors, though technically callable
+//! from Solana programs, are not defined somewhere accessible to Solana
+//! programs. This is a historical artifact due to lack of use cases for
+//! invocation from other programs.
+//!
+//! Locations of program IDs and instruction constructors are noted in the lists
+//! below.
+//!
+//! While some native programs have been active since the genesis block, others
+//! are activated dynamically after a specific slot, and some are not yet
+//! active. This documentation does not distinguish which native programs are
+//! actually active on any particular network. The `solana feature status` CLI
+//! command can help in determining active features.
+//!
+//! The native programs are:
+//!
+//! ## Built-ins
+//!
+//! - __System Program__: Creates new accounts, allocates account data, assigns
+//!   accounts to owning programs, transfers lamports from System Program owned
+//!   accounts and pays transaction fees.
+//!   - ID: [`solana_program::system_program`]
+//!   - Instruction: [`solana_program::system_instruction`]
+//!
+//! - __Stake Program__: Creates and manages accounts representing stake and
+//!   rewards for delegations to validators.
+//!   - ID: [`solana_program::stake::program`]
+//!   - Instruction: [`solana_program::stake::instruction`]
+//!
+//! - __Vote Program__: Creates and manage accounts that track validator voting
+//!   state and rewards.
+//!   - ID: [`solana_program::vote::program`]
+//!   - Instruction: [`solana_vote_program::vote_instruction`](https://docs.rs/solana-vote-program/latest/solana_vote_program/vote_instruction/index.html)
+//!
+//! - __Config Program__: Adds configuration data to the chain and the list of
+//!   public keys that are permitted to modify it.
+//!   - ID: [`solana_program::config::program`]
+//!   - Instruction: [`solana_config_program::config_instruction`](https://docs.rs/solana-config-program/latest/solana_config_program/config_instruction/index.html)
+//!
+//! - __Address Lookup Table Program__: Support for [on-chain address lookup tables][lut].
+//!   - ID: [`solana_address_lookup_table_program`](https://docs.rs/solana-address-lookup-table-program/latest/solana_address_lookup_table_program/)
+//!   - Instruction: [`solana_address_lookup_table_program::instruction`](https://docs.rs/solana-address-lookup-table-program/latest/solana_address_lookup_table_program/instruction/index.html)
+//!
+//! [lut]: https://docs.solana.com/proposals/transactions-v2
+//!
+//! ## Special built-ins
```

Comment:
The compute budget program can probably just be lumped into the precompiled list.  And actually we could probably refer to these as all builtins and then specify which are program invokable and which are only supported as top-level instructions